packages/vuelidate/src/core.js

```diff
@@ -303,26 +294,47 @@ function createValidationResults (rules, state, key, parentKey, resultsCache, pa
   return result
 }
 
+/**
+ * create protected state for cases when the state branch does not exist yet.
+ * This protects when using the OptionsAPI as the data is LINKED after the setup
+ * method
+ * @param {Object} state
+ * @param {string} key
+ * @returns {Ref<*>>|ComputedRef<*>}
+ */
+function getNestedState (state, key) {
+  const s = unwrap(state)
+
+  if (!s) {
+    return computed(() => {
+      const s = unwrap(state)
+      return s ? s[key] : undefined
+    })
+  }
+
+  if (isRef(s[key]) || isReactive(s[key])) return s[key]
+
+  return computed(() => {
+    return unwrap(state)[key]
+  })
```

Comment:
The multiple instances of `s` and the illusion of duplicated code due to use of `computed` makes this a bit hard to read. Maybe flipping the return statements would let us consolidate?

```suggestion
  const initialState = unwrap(state)

  if (initialState && (isRef(initialState[key]) || isReactive(initialState[key]))) return initialState[key]

  // If state isn't available during setup, created a computed that will update when becomes available
  return computed(() => {
      const s = unwrap(state)
      return s ? s[key] : undefined
  })
```